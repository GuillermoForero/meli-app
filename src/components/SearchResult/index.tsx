import Box from 'components/Box';
import React from 'react';
import ic_ML from 'assets/Logo_ML2x.png'
import { COLORS } from 'consts';
import ResultItem from 'components/ResultItem';

const SearchResult: React.FC = () => (
  <Box alignItems="center" justifyContent="center" flexDirection="column" width="80%" margin="40px 0 0" style={{ backgroundColor: `${COLORS.WHITE}` }}>
    <ResultItem imgSrc={ic_ML} price="20000" description="Descripción del producto" city="mendoza" isShipping={true}/>
    <ResultItem imgSrc={ic_ML} price="20000" description="Descripción del producto" city="mendoza" isShipping={true}/>
    <ResultItem imgSrc={ic_ML} price="20000" description="Descripción del producto" city="mendoza" isShipping={true}/>
    <ResultItem imgSrc={ic_ML} price="20000" description="Descripción del producto" city="mendoza" isShipping={true}/>
    <ResultItem imgSrc={ic_ML} price="20000" description="Descripción del producto" city="mendoza" isShipping={true}/>
  </Box>
);

export default SearchResult;