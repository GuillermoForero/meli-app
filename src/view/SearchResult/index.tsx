import Box from 'components/Box';
import React from 'react';
import { COLORS } from 'consts';
import ResultItem from 'view/SearchResult/components/ResultItem';
import { itemsSelectors } from 'state/items';
import { useSelector } from 'react-redux';

const SearchResult: React.FC = () => {
  const searchItems = useSelector(itemsSelectors.searchItems);
  return (
    <Box alignItems="center" justifyContent="center" flexDirection="column" width="80%" margin="40px 0 0" style={{ backgroundColor: `${COLORS.WHITE}` }}>
      {searchItems?.items.map((item, index) => (
        <ResultItem key={index} imgSrc={item.picture} price={item.price} title={item.title} city="mendoza" freeShipping={item.free_shipping} />  
      ))}
    </Box>
  )
};

export default SearchResult;