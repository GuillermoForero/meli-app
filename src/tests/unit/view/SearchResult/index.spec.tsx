import React from 'react';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import { render } from 'tests/support/customRender';
import { appStoreFull } from 'tests/support/stubAppStore';
import SearchResult from 'view/SearchResult';

describe('[SearchResult]', () => {
  const mockStore = createMockStore();
  it('Renders default snapshot', () => {
    const store = mockStore(appStoreFull);
    const { container, getByText } = render(
      <Provider store={store}>
        <SearchResult />
      </Provider>);
    expect(getByText('title_1')).toBeDefined();
    expect(getByText('mendoza')).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
