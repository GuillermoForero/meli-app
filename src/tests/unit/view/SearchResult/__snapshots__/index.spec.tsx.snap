// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SearchResult] Renders default snapshot 1`] = `
<div>
  <div
    class="sc-bdfBwQ bjkWJX"
    width="100%"
  >
    <div
      class="ant-row"
      style="width: 100%;"
    >
      <div
        class="ant-col ant-col-2"
      />
      <div
        class="ant-col ant-col-20"
        style="background-color: white;"
      >
        <div
          class="sc-bdfBwQ kqamos"
          height="250px"
          style="background-color: white; border-bottom: 1px solid #EEEEEE; cursor: pointer;"
        >
          <div
            class="sc-bdfBwQ cfUgem"
            height="100%"
            width="auto"
          >
            <img
              alt="itemImg"
              class="sc-dlfnbm jOTYEy"
              src="url"
            />
            <div
              class="sc-bdfBwQ dzmWob"
              width="auto"
            >
              <div
                class="sc-bdfBwQ jCueMc"
              >
                <p
                  class="sc-gsTCUz cCdcQx"
                  color="GRAY_20"
                  font-size="24px"
                >
                  $ 
                  price_1
                </p>
                <img
                  alt="shippingImg"
                  class="sc-hKgILt fzDEKB"
                  src="ic_shipping2x.png"
                />
              </div>
              <h4
                class="sc-gsTCUz fCxHZK"
                color="GRAY_40"
                font-size="18px"
              >
                title_1
              </h4>
            </div>
          </div>
          <div
            class="sc-bdfBwQ iHhrNG"
            width="auto"
          >
            <span
              class="sc-gsTCUz eIfkPM"
              color="GRAY_60"
              font-size="12px"
            >
              mendoza
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
